### PR TITLE
feat: init github-runner-host and github-runner container

### PR DESCRIPTION
**Pull request by steveej**

### CODE REVIEW

File: nixos-hosts/github-runner-host/configuration.nix [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  # Or disable the firewall altogether.
+  networking.firewall.enable = true;
+
+  # Copy the NixOS configuration file and link it from the resulting system
+  # (/run/current-system/configuration.nix). This is useful in case you
+  # accidentally delete configuration.nix.
+  # system.copySystemConfiguration = true;
+
+  containers = {
+    github-runner.config = github-runner.config;
```
Comment (colemickens) [inline nixos-hosts/github-runner-host/configuration.nix]: I'd recommend doing something more like `github-runner = inputs.self.outputs.nixosConfigurations.containers.github-runner`.

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  description = "An example NixOS configuration";
+
+  inputs = {
+    nixpkgs = { url = "github:nixos/nixpkgs/nixos-unstable"; };
+    nixpkgs-github-runner = { url = "github:codedownio/nixpkgs/multiple-github-runners"; };
+  };
+
+  outputs = inputs: {
+    nixosConfigurations = rec {
+      containers.github-runner = inputs.nixpkgs-github-runner.lib.nixosSystem  {
```
Comment (colemickens) [inline flake.nix]: seems odd to use someone else's lib to get a nixosSystem when you already have it from your nixpkgs import.

In fact, I'm not sure what other import is even for in this case? Maybe it's not utilized fully yet?

**Review by colemickens — CHANGES_REQUESTED** [review]

File: flake.nix (hunk 2 above)
Comment (colemickens) [reply flake.nix]: Hm, I see, you want the other thing's package set to make up the container. Seems a bit weird to me, but ok.

Comment (steveej) [reply flake.nix]: the rationale is to use the nixpkgs-github-runner input for the container config. it's different from the upstream nixpkgs

Comment (colemickens) [reply flake.nix]: Oh... I see, it's probably just a nixpkgs fork with some fixes or something. Got it.

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      };
+
+      github-runner-host = inputs.nixpkgs.lib.nixosSystem {
+        system = "x86_64-linux";
+        modules = [
+          ./nixos-hosts/github-runner-host/configuration.nix
+        ];
+
+        specialArgs = { 
+          github-runner = containers.github-runner;
```
Comment (colemickens) [inline flake.nix]: Again, not necessarily wrong to do this, but I think you'll find yourself looking like more flakes if you only pass in `inputs` and then access your self outputs through `inputs.self`.

**Comment by steveej** [comment]
many thanks @colemickens :wave: :bowing_man:

**Pull request closed without merging**